test/e2e/sched/uninstall/uninstall.go

```diff
@@ -0,0 +1,35 @@
+package uninstall
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	e2eclient "github.com/openshift-kni/numaresources-operator/test/utils/clients"
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects"
+)
+
+var _ = Describe("[Scheduler] uninstall", func() {
+	Context("with a running cluster with all the components", func() {
+		It("should delete all components after NROScheduler deletion", func() {
+			By("deleting the NROScheduler object")
+			nroSchedObj := objects.TestNROScheduler()
+
+			// failed to get the NROScheduler object, nothing else we can do
+			if err := e2eclient.Client.Get(context.TODO(), client.ObjectKeyFromObject(nroSchedObj), nroSchedObj); err != nil {
+				if !errors.IsNotFound(err) {
+					klog.Warningf("failed to get the NUMA resource scheduler %q: %w", nroSchedObj.Name, err)
+				}
+				return
+			}
+
+			err := e2eclient.Client.Delete(context.TODO(), nroSchedObj)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
ok, and now we should check no dependent objects are left in the namespace, shouldn't we?